src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -291,11 +299,17 @@ private void ProcessServerDescriptionChanged(ServerDescriptionChangedEventArgs a
                 }
                 else
                 {
+#pragma warning disable 618
                     if (IsServerValidForCluster(newClusterDescription.Type, Settings.ConnectionMode, newServerDescription.Type))
+#pragma warning restore 618
                     {
                         if (newClusterDescription.Type == ClusterType.Unknown)
                         {
-                            newClusterDescription = newClusterDescription.WithType(newServerDescription.Type.ToClusterType());
+                            if (newServerDescription.Type != ServerType.ReplicaSetGhost)
```

Comment:
This change is related to this spec line: https://github.com/mongodb/specifications/blob/e56f5eceed7729f8b9b43a4a1f76c7e5840db49f/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#topologytype-remains-unknown-when-an-rsghost-is-discovered